spec/lib/alaveteli_geoip_spec.rb

```diff
@@ -2,119 +2,156 @@
 require File.expand_path(File.dirname(__FILE__) + '/../spec_helper')
 
 describe AlaveteliGeoIP do
+  let(:ip_address) { '127.0.0.1' }
+
   describe '.country_code_from_ip' do
+    subject { described_class.country_code_from_ip }
+
     it 'delegates to an instance of the class' do
-      instance = double
-      allow(AlaveteliGeoIP).to receive(:instance).and_return(instance)
-      expect(instance).to receive(:country_code_from_ip).with('127.0.0.1')
-      AlaveteliGeoIP.country_code_from_ip '127.0.0.1'
+      delegate = double(described_class)
+      allow(described_class).to receive(:instance).and_return(delegate)
+      expect(delegate).to receive(:country_code_from_ip).with(ip_address)
+      AlaveteliGeoIP.country_code_from_ip(ip_address)
     end
   end
 
   describe '.instance' do
-    it 'creates a new instance' do
-      expect(AlaveteliGeoIP.instance).to be_instance_of(AlaveteliGeoIP)
-    end
+    subject { described_class.instance }
+
+    it { is_expected.to be_a(described_class) }
 
     it 'caches the instance' do
       expect(AlaveteliGeoIP.instance).to equal(AlaveteliGeoIP.instance)
     end
   end
 
   describe '.new' do
+    subject { described_class.new }
+
     it 'configures the instance with the configured country code' do
-      expect(AlaveteliGeoIP.new.current_code).
+      expect(subject.current_code).
         to eq(AlaveteliConfiguration.iso_country_code)
     end
 
     context 'if a database param is supplied' do
-      it 'configures the instance with the database specified' do
+      before do
         allow(File).to receive(:file?).
           with('/my/geoip/database').and_return(true)
+      end
+
+      it 'configures the instance with the database specified' do
         expect(GeoIP).to receive(:new).with('/my/geoip/database')
-        AlaveteliGeoIP.new('/my/geoip/database')
+        described_class.new('/my/geoip/database')
       end
     end
 
     context 'if there is a geoip database configured and present' do
-      it 'configures the instance with an instance of geoip' do
+      before do
         allow(File).to receive(:file?).
           with(AlaveteliConfiguration.geoip_database).and_return(true)
+      end
+
+      it 'configures the instance with an instance of geoip' do
         expect(GeoIP).to receive(:new).
           with(AlaveteliConfiguration.geoip_database)
-        AlaveteliGeoIP.new
+        subject
       end
     end
```

Comment:
There is no spec checking if `GeoIP.new...` is assigned to `subject.geoip` as there is for `subject.gaze_url` below. Perhaps we should add one?